hack/quickstart-setup.sh

```diff
@@ -16,6 +16,39 @@
 # limitations under the License.
 #
 
+set -e pipefail
+
+containerRuntime() {
+  local container_runtime=""
+  if command -v podman &>/dev/null; then
+    container_runtime="podman"
+  elif command -v docker &>/dev/null; then
+    container_runtime="docker"
+  else
+    echo "Neither Docker nor Podman is installed. Exiting..."
+    exit 1
+  fi
+  echo "$container_runtime"
+}
```

Comment:
Why favour podman? Kind will favour docker afaik, so if you have both installed it will create the kind cluster using docker and this will try and use podman.

Reply:
https://github.com/kubernetes-sigs/kind/blob/main/pkg/cluster/provider.go#L114

Reply:
yeah, we could favour docker now

There was a previous version of this that had some podman specifics, and since a bunch of podman users alias docker to podman, these commands would fail. the networking commands in particular (hence the ordering).

those are gone now, and the MetalLB address pool stuff doesn't use `$CONTAINER_RUNTIME_BIN`, so this could be reverted

Reply:
docker default now

Reply:
I think to realistically make this work in all scenarios you need to check more than just whether or not the commands exist, and also it needs to match the behaviour of the kind provider auto detection, where it checks docker first.

Regardless of the order, any of these could be setup as an alias, so you need to check docker is really docker and the same with podman. You can see here with kind they also check the version string contains the expected runtime name https://github.com/kubernetes-sigs/kind/blob/main/pkg/cluster/internal/providers/docker/util.go#L33. That would fail the docker check if it was actually podman, where this currently wouldn't.

All that aside, the only reason we care about docker vs podman is because of the network setup that is slightly different right? Everything else is the same? If so, do we really need any of this if you make the network check more about what is being returned from the container runtime network inspect command, rather than the command running it.

```
set +e
containerRuntime=docker
SUBNET=`${containerRuntime} network inspect $networkName -f '{{ (index .IPAM.Config 0).Subnet }}'`
if [[ -z "$SUBNET" ]]; then
  SUBNET=`${containerRuntime} network inspect -f '{{range .Subnets}}{{if eq (len .Subnet.IP) 4}}{{.Subnet}}{{end}}{{end}}' $networkName`
fi

if [[ -z "$SUBNET" ]]; then
   echo "Error: parsing IPv4 network address for '$networkName' docker network"
   exit 1
fi
set -e
```

We could just have it use `docker` as the container runtime command all the time and if you want to use podman, setup an alias. If you don't want to setup an alias we could just use the kind provider override env vars to determine if its docker or podman KIND_EXPERIMENTAL_PROVIDER=podman.